pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -155,6 +155,17 @@ func buildHTTPRoute(namespace, urlPath string, appNames []string) ([]networking.
 			Rewrite: &networking.HTTPRewrite{
 				Authority: network.GetServiceHostname(appName, namespace),
 			},
+			Headers: &networking.Headers{
+				Request: &networking.HeaderOperations{
+					Set: map[string]string{
+						// Set forwarding headers so the app gets the real hostname it's serving
+						// at rather than the internal one:
+						// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded1
+						"X-Forwarded-Host": hostDomain,
```

Comment:
I think this should be using `Add` instead of `Set`. That way the values are comma-separated instead of replacing existing values: https://godoc.org/knative.dev/pkg/apis/istio/v1alpha3#HeaderOperations